crates/ignore/src/gitignore.rs

```diff
@@ -389,7 +390,12 @@ impl GitignoreBuilder {
             Err(err) => return Some(Error::Io(err).with_path(path)),
             Ok(file) => file,
         };
-        let rdr = io::BufReader::new(file);
+        let rdr = io::BufReader::new(
+            DecodeReaderBytesBuilder::new()
+                .bom_sniffing(false)
```

Comment:
I'm not sure if we actually want to enable bom_sniffing here. `bom_sniffing(false)` seems to be behavior preserving to me. But maybe we want to support ignore files in utf-16 as well?